ores/score_caches/redis.py

```diff
@@ -81,3 +84,84 @@ def from_config(cls, config, name, section_key="score_caches"):
     def hash_cache(cls, cache):
         sorted_tuple = tuple(sorted(cache.items()))
         return sha1(bytes(str(sorted_tuple), 'utf8')).hexdigest()
+
+
+class RedisSentinel(ScoreCache):
+
+    def __init__(self, sentinel, ttl=None, prefix=None, cluster=None, socket_timeout=None):
+        self.sentinel = sentinel
+        self.ttl = int(ttl or TTL)
+        self.prefix = str(prefix or PREFIX)
+        self.cluster = str(cluster or CLUSTER)
+        self.socket_timeout = float(socket_timeout or SOCKET_TIMEOUT)
+
+    def lookup(self, context_name, model_name, rev_id, version=None,
+               injection_cache=None):
+        key = self._generate_key(
+            context_name, model_name, rev_id, version=version,
+            injection_cache=injection_cache)
+
+        replica = self.sentinel.slave_for(self.cluster, socket_timeout=self.socket_timeout)
+        sentinel_logger.debug("Looking up score at {0} in replica".format(key))
+        value = replica.get(key)
+        if value is None:
+            raise KeyError(key)
+        else:
+            return json.loads(str(value, 'utf-8'))
+
+    def store(self, score, context_name, model_name, rev_id, version=None,
+              injection_cache=None):
+        key = self._generate_key(
+            context_name, model_name, rev_id, version=version,
+            injection_cache=injection_cache)
+
+        master = self.sentinel.master_for(self.cluster, socket_timeout=self.socket_timeout)
+        sentinel_logger.debug("Storing score at {0} in master".format(key))
+        master.setex(key, self.ttl, bytes(json.dumps(score), 'utf-8'))
```

Comment:
For example, Redis and RedisSentinel could vary only here, so `store` might be shared but `_setex` would be overridden in the subclass.